python_modules/libraries/dagster-sdf/dagster_sdf/information_schema.py

```diff
@@ -0,0 +1,207 @@
+import os
+from pathlib import Path
+from typing import Dict, List, Mapping, Optional, Set, Tuple, Union
+
+import pandas as pd
+from dagster import AssetKey, AssetOut, Nothing, TableColumn, TableSchema
+from dagster._core.definitions.metadata import (
+    CodeReferencesMetadataSet,
+    CodeReferencesMetadataValue,
+    LocalFileCodeReference,
+    TableColumnConstraints,
+    TableMetadataSet,
+)
+from dagster._core.definitions.tags import StorageKindTagSet
+from pydantic import BaseModel, DirectoryPath
+
+from .asset_utils import dagster_name_fn, default_asset_key_fn
+from .constants import (
+    DEFAULT_SDF_WORKSPACE_ENVIRONMENT,
+    SDF_INFORMATION_SCHEMA_TABLES,
+    SDF_TARGET_DIR,
+)
+from .errors import DagsterSdfInformationSchemaNotFoundError
+
+
+class LazyDataFrameDict:
```

Comment:
Could you write a brief docstring on why this is necessary? How large are these dataframes that we need to do this in a lazy fashion?

Reply:
Dataframes should not be too large, but for large workspaces with complex dags they could be in the range of 50mb and over is my guess